involute_gears.scad

```diff
@@ -42,488 +42,538 @@ pi=3.1415926535897932384626433832795;
+    
+    // Variables controlling the bore
+    bore_diameter = (bore_diameter!=undef?bore_diameter:root_radius * .1);
+
+    // Variables controlling the circular holes in the gear.
+    circles = spokes==0?circles:0;
+    circle_orbit_diameter=hub_diameter/2+rim_radius;
+    circle_orbit_curcumference=pi*circle_orbit_diameter;
+
+    // Limit the circle size to 90% of the gear face.
+    circle_diameter=
+        min (
+            0.70*circle_orbit_curcumference/circles,
+            (rim_radius-hub_diameter/2)*0.9);
+
+    difference()
+    {
+        union ()
+        {
+            difference ()
+            {
+                //start with a plane toothed disk gear
+                linear_extrude_flat_option(flat=flat, height=rim_thickness, convexity=10, twist=twist)
+                gear_shape (
+                    number_of_teeth,
+                    pitch_radius = pitch_radius,
+                    root_radius = root_radius,
+                    base_radius = base_radius,
+                    outer_radius = outer_radius,
+                    half_thick_angle = half_thick_angle,
+                    involute_facets=involute_facets);
+
+                //if we have a spoked gear, then the gear body must be removed.
+                if (spokes>0)
+                    translate ([0,0,-1])
+                    cylinder (r=rim_radius,h=rim_thickness+2);
+                //if the rim is thicker than the gear, carve out the gear body
+                else if (rim_thickness>gear_thickness)
+                    translate ([0,0,gear_thickness])
+                        cylinder (r=rim_radius,h=rim_thickness);
+            }
+            
+            //extend the gear body if gear_thickness > rim_thickness unless spoked, 
+            if (spokes==0 && gear_thickness > rim_thickness)
+            {
+                linear_extrude_flat_option(flat=flat, height=gear_thickness)
+                //if rim is thicker than body, body protrudes into rim
+                circle (r=rim_radius);
+            }
+            //add the hub
+            linear_extrude_flat_option(flat=flat, height=hub_thickness)
+                circle (r=hub_diameter/2);
+            
+            //add in spokes
+            if (spokes>0) 
+                for(i=[0:spokes-1])
+                    translate([0,0,rim_thickness/2])
+                    rotate([90,0,i*360/spokes])
+                    translate([0,0,spoke_radius])
+                    {
+                        if (spoke_square==true){
+                             resize([spoke_width,spoke_thickness,spoke_length])
+                            translate([0,0,.5])
+                            cube(1,center=true);
+                        }
+                        if (spoke_square==false){
+                            resize([spoke_width,spoke_thickness,spoke_length])
+                            cylinder(h=1,d=1);
```

Comment:
I played around with the `spokes` parameter for a bit and noticed that the spokes have very very few facets. Since you're using `resize()` on a cylinder, how about starting from a larger cylinder? I found that starting from `cylinder(h=10, d=10)` worked a lot better.

Reply:
will fix